File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        unordered_map<int, int> mp;
        for (int i = 0; i < nums.size(); i++)
        {
            int key = target - nums[i];
            if (mp.count(key))
            {
                return {mp.at(key), i};
            }
            mp.insert({nums[i], i});
        }   
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        L, R = 0, len(numbers) - 1
        while L < R:
            if target > numbers[L] + numbers[R]:
                L += 1
            elif target < numbers[L] + numbers[R]:
                R -= 1
            else:
                return [L + 1, R + 1]
        return [L, R]